internal/matchers/signature.go

```diff
@@ -21,6 +21,30 @@ type (
 	}
 )
 
+// P7s matches an .p7s signature File (PEM,Base64)
+func P7s(in []byte) bool {
+	// Check for PEM Encoding
+	if bytes.Contains(in, []byte("BEGIN PKCS7")) {
+		return true
+	}
+	// Check if DER Encoding is long enough
+	if len(in) < 20 {
+		return false
+	}
+	// Magic Bytes for the signedData ASN.1 encoding
+	startHeader := [][]byte{{0x30, 0x80}, {0x30, 0x81}, {0x30, 0x82}, {0x30, 0x83}, {0x30, 0x84}}
+	signedDataMatch := []byte{0x26, 0x06, 0x09, 0x2A, 0x86, 0x48, 0x86, 0xF7, 0x0D, 0x01, 0x07}
+	// Check if Header is correct. There are multiple valid headers
+	for _, match := range startHeader {
+		if bytes.Contains(in[0:2], match) { // If first bytes match -> check for ASN.1 Object Type
+			if bytes.Contains(in[2:20], signedDataMatch) { // only check first part of the ASN.1 Message for OBJECT: signedData
```

Comment:
The check for DER described in [tika](https://github.com/apache/tika/blob/75c2ff5686a70c0fb15c4b52534c1be09669af1b/tika-core/src/main/resources/org/apache/tika/mime/tika-mimetypes.xml#L727) is a little different.
When the first two bytes are 0x3080, the 0x06092a864886f70d0107FFa0 bytes are searched for at index 2 in the input.
But if the first two bytes are 0x3081,   the 0x06092a864886f70d0107FFa0 bytes are searched for at offset 3 in the input.

Your code is always checking in[2:20], but it should check 
 - in[3:21] for {0x30, 0x81},
 - in[4:22] for {0x30, 0x82}, and so on

Reply:
Implementing `i` as a counter value for the offset in the loop should be sufficient